test-adapter/main.go

```diff
@@ -56,13 +55,12 @@ func (a *SampleAdapter) makeProviderOrDie() (provider.MetricsProvider, *restful.
 func main() {
 	logs.InitLogs()
 	defer logs.FlushLogs()
-	klog.InitFlags(nil)
 
 	cmd := &SampleAdapter{}
 	cmd.Name = "test-adapter"
 
 	cmd.Flags().StringVar(&cmd.Message, "msg", "starting adapter...", "startup message")
-	cmd.Flags().AddGoFlagSet(flag.CommandLine) // make sure we get the klog flags
+	logs.AddFlags(cmd.Flags())
```

Comment:
Instead of having that globally defined in the main it would be better to add a Logging config to the server config based on the component-base logs package (https://pkg.go.dev/k8s.io/component-base/logs),

Reply:
I'm not sure I understand your request.
Here, I'm already using the component-base logs package, to add logging flags.

Do you suggest configuring the logging in the library (i.e. removing it from test-adapter)?

Reply:
Sorry for not explaining my suggestion properly, I was thinking about putting the Logging configure inside of the `CustomMetricsAdapterServerOptions` structure, similarly to https://github.com/kubernetes-sigs/metrics-server/blob/master/cmd/metrics-server/app/options/options.go#L46.

Reply:
Ok, I understand, thank you. However, metrics-server is a server, and this is a library. I'm not sure the _library_ `custom-metrics-apiserver` should be concerned by logging, and should even depend on `klog` and `k8s.io/component-base/logs`.

Actually, the only klog usage in the library (i.e. outside the test-adapter implementation) is: https://github.com/kubernetes-sigs/custom-metrics-apiserver/blob/c435b4276bc46728e4b4b4c1f0a3f981d91d0815/pkg/dynamicmapper/mapper.go#L46

(perhaps this log entry should be removed. Or on the contrary, perhaps more logging should be added in the library?)

That's why I preferred keeping the logging stuff in test-adapter, just to provide a working _example_ implementation.

Do you still prefer that I add logging settings (and flags) in the `CustomMetricsAdapterServerOptions`?

Reply:
Ah I wasn't aware that we didn't have much logging in the library, I never looked for that particularly before.

Let's keep the log entry for now and we will most likely revisit it once we bring the resource metrics API over and start unifying the implementations.

> Do you still prefer that I add logging settings (and flags) in the CustomMetricsAdapterServerOptions?

No, now that you've mentioned that we don't log a lot of things in the library I don't think we should do that right now, maybe in the future though.

Reply:
> Let's keep the log entry for now and we will most likely revisit it once we bring the resource metrics API over and start unifying the implementations.

Ok, thanks